subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -991,9 +970,16 @@ public ArtifactView artifactView() {
 
 
             @Override
-            public ArtifactView attributes(Map<?, ?> attributeMap) {
+            public AttributeContainer getAttributes() {
+                return getViewAttributes();
+            }
+
+            @Override
+            public ArtifactView attributes(Action<? super AttributeContainer> action) {
                 assertUnset("attributes", viewAttributes);
-                this.viewAttributes = attributesFactory.fromPolymorphicMap(attributeMap);
+                DefaultMutableAttributeContainer mutableAttributes = new DefaultMutableAttributeContainer(attributesFactory);
```

Comment:
To be consistent, `getAttributes()` should also always return the mutable container before `getFiles` or `getArtifacts` is called. Atm it returns `EMPTY` for an empty container.